### ISSUE TITLE
Schema compilation still failing for schema version > 0.2.3

### ISSUE THREAD

**Issue by yatharthranjan** [RADAR-base/RADAR-Docker#116]
Schema compilation is failing with following log output - 

```
sudo docker-compose run --rm kafka-init
Starting radarcphadoopstack_zookeeper-1_1 ... done
Starting radarcphadoopstack_kafka-1_1 ... done
Starting radarcphadoopstack_kafka-2_1 ... done
Starting radarcphadoopstack_kafka-3_1 ... done
Starting radarcphadoopstack_schema-registry-1_1 ... done
Compiling schemas...
Input files to compile:
  merged/commons/stream/aggregator/phone_usage_aggregate.avsc
  merged/commons/stream/aggregator/aggregate_list.avsc
  merged/commons/stream/aggregator/numeric_aggregate.avsc
  merged/commons/monitor/application/application_uptime.avsc
  merged/commons/monitor/application/application_record_counts.avsc
  merged/commons/monitor/application/application_server_status.avsc
  merged/commons/monitor/application/application_external_time.avsc
  merged/commons/active/questionnaire/questionnaire.avsc
  merged/commons/active/thincit/thinc_it_pdq.avsc
  merged/commons/active/thincit/thinc_it_input_type.avsc
  merged/commons/active/thincit/thinc_it_trails.avsc
  merged/commons/active/thincit/thinc_it_symbol_check.avsc
  merged/commons/active/thincit/thinc_it_spotter.avsc
  merged/commons/active/thincit/thinc_it_code_breaker.avsc
  merged/commons/kafka/aggregate_key.avsc
  merged/commons/kafka/observation_key.avsc
  merged/commons/catalogue/processing_state.avsc
  merged/commons/catalogue/radar_widget.avsc
  merged/commons/catalogue/time_window.avsc
  merged/commons/catalogue/unit.avsc
  merged/commons/passive/weather/local_weather.avsc
  merged/commons/passive/pebble/pebble2_battery_level.avsc
  merged/commons/passive/pebble/pebble2_acceleration.avsc
  merged/commons/passive/pebble/pebble2_heart_rate.avsc
  merged/commons/passive/pebble/pebble2_heart_rate_filtered.avsc
  merged/commons/passive/phone/phone_step_count.avsc
  merged/commons/passive/phone/phone_acceleration.avsc
  merged/commons/passive/phone/phone_sms.avsc
  merged/commons/passive/phone/phone_call.avsc
  merged/commons/passive/phone/phone_battery_level.avsc
  merged/commons/passive/phone/phone_bluetooth_devices.avsc
  merged/commons/passive/phone/phone_relative_location.avsc
  merged/commons/passive/phone/phone_user_interaction.avsc
  merged/commons/passive/phone/phone_light.avsc
  merged/commons/passive/phone/phone_gyroscope.avsc
  merged/commons/passive/phone/phone_usage_event.avsc
  merged/commons/passive/phone/phone_magnetic_field.avsc
  merged/commons/passive/phone/phone_sms_unread.avsc
  merged/commons/passive/phone/phone_contact_list.avsc
  merged/commons/passive/empatica/empatica_e4_battery_level.avsc
  merged/commons/passive/empatica/empatica_e4_electrodermal_activity.avsc
  merged/commons/passive/empatica/empatica_e4_tag.avsc
  merged/commons/passive/empatica/empatica_e4_acceleration.avsc
  merged/commons/passive/empatica/empatica_e4_blood_volume_pulse.avsc
  merged/commons/passive/empatica/empatica_e4_inter_beat_interval.avsc
  merged/commons/passive/empatica/empatica_e4_sensor_status.avsc
  merged/commons/passive/empatica/empatica_e4_temperature.avsc
  merged/commons/passive/biovotion/biovotion_vsm1_ppg_raw.avsc
  merged/commons/passive/biovotion/biovotion_vsm1_led_current.avsc
  merged/commons/passive/biovotion/biovotion_vsm1_galvanic_skin_response.avsc
  merged/commons/passive/biovotion/biovotion_vsm1_temperature.avsc
  merged/commons/passive/biovotion/biovotion_vsm1_blood_pulse_wave.avsc
  merged/commons/passive/biovotion/biovotion_vsm1_battery_level.avsc
  merged/commons/passive/biovotion/biovotion_vsm1_heart_rate_variability.avsc
  merged/commons/passive/biovotion/biovotion_vsm1_energy.avsc
  merged/commons/passive/biovotion/biovotion_vsm1_oxygen_saturation.avsc
  merged/commons/passive/biovotion/biovotion_vsm1_acceleration.avsc
  merged/commons/passive/biovotion/biovotion_vsm1_respiration_rate.avsc
  merged/commons/passive/biovotion/biovotion_vsm1_heart_rate.avsc
log4j:WARN No appenders could be found for logger (AvroVelocityLogChute).
log4j:WARN Please initialize the log4j system properly.
log4j:WARN See http://logging.apache.org/log4j/1.2/faq.html#noconfig for more info.
Exception in thread "main" org.apache.avro.SchemaParseException: Undefined name: "NumericAggregate"
	at org.apache.avro.Schema.parse(Schema.java:1228)
	at org.apache.avro.Schema.parse(Schema.java:1306)
	at org.apache.avro.Schema.parse(Schema.java:1269)
	at org.apache.avro.Schema$Parser.parse(Schema.java:1032)
	at org.apache.avro.Schema$Parser.parse(Schema.java:997)
	at org.apache.avro.tool.SpecificCompilerTool.run(SpecificCompilerTool.java:92)
	at org.apache.avro.tool.Main.run(Main.java:87)
	at org.apache.avro.tool.Main.main(Main.java:76)
```

Interestingly this only fails for schema version 0.3 or greater. I guess that is when `NumericAggregate` was added. 
Also this is only failing for production server even after the dependent enum fix in `kafka-radarinit/init.sh`. It works fine on amazon test instance. 
Any thoughts @blootsvoets , @nivemaham ?

**Comment by nivemaham** [comment]
Yes. You are right that it was added on version 0.3.0. But i could see the [schema](https://github.com/RADAR-base/RADAR-Schemas/blob/v0.3/commons/stream/aggregator/numeric_aggregate.avsc) in the repository and the the radar-schemas-commons also has the NumricAggregate class.

**Comment by nivemaham** [comment]
Right. So it works for test env. Did we also have older versions of schema and corresponding streams and topics registered in amazon test env? Could it be that the schema updates didn't meet the schema-compatibility?

**Comment by yatharthranjan** [comment]
So apparently the inverse Regex is not working as intended, its because even in dependent files there are lines which do not contain capital dependent classes, so it just includes all the schemas (dependent as well as not dependent) in the $notdependent array. Look below for example - 
```json
Not Dependent: merged/commons/stream/aggregator/phone_usage_aggregate.avsc merged/commons/stream/aggregator/aggregate_list.avsc merged/commons/stream/aggregator/numeric_aggregate.avsc merged/commons/stream/source_statistics.avsc merged/commons/monitor/application/application_time_zone.avsc merged/commons/monitor/application/application_uptime.avsc merged/commons/monitor/application/application_record_counts.avsc merged/commons/monitor/application/application_server_status.avsc merged/commons/monitor/application/application_external_time.avsc merged/commons/active/opensmile/open_smile_2_audio_analysis.avsc merged/commons/active/opensmile/open_smile_2_audio_recording.avsc merged/commons/active/notification/notification.avsc merged/commons/active/task/task.avsc merged/commons/active/questionnaire/questionnaire.avsc merged/commons/active/thincit/thinc_it_pdq.avsc merged/commons/active/thincit/thinc_it_input_type.avsc merged/commons/active/thincit/thinc_it_trails.avsc merged/commons/active/thincit/thinc_it_symbol_check.avsc merged/commons/active/thincit/thinc_it_spotter.avsc merged/commons/active/thincit/thinc_it_code_breaker.avsc merged/commons/kafka/aggregate_key.avsc merged/commons/kafka/observation_key.avsc merged/commons/catalogue/processing_state.avsc merged/commons/catalogue/time_window.avsc merged/commons/catalogue/unit.avsc merged/commons/passive/bittium/bittium_faros_temperature.avsc merged/commons/passive/bittium/bittium_faros_inter_beat_interval.avsc merged/commons/passive/bittium/bittium_faros_battery_level.avsc merged/commons/passive/bittium/bittium_faros_ecg.avsc merged/commons/passive/bittium/bittium_faros_acceleration.avsc merged/commons/passive/weather/local_weather.avsc merged/commons/passive/pebble/pebble2_battery_level.avsc merged/commons/passive/pebble/pebble2_acceleration.avsc merged/commons/passive/pebble/pebble2_heart_rate.avsc merged/commons/passive/pebble/pebble2_heart_rate_filtered.avsc merged/commons/passive/phone/phone_step_count.avsc merged/commons/passive/phone/phone_acceleration.avsc merged/commons/passive/phone/phone_sms.avsc merged/commons/passive/phone/phone_call.avsc merged/commons/passive/phone/phone_battery_level.avsc merged/commons/passive/phone/phone_bluetooth_devices.avsc merged/commons/passive/phone/phone_relative_location.avsc merged/commons/passive/phone/phone_user_interaction.avsc merged/commons/passive/phone/phone_light.avsc merged/commons/passive/phone/phone_gyroscope.avsc merged/commons/passive/phone/phone_usage_event.avsc merged/commons/passive/phone/phone_magnetic_field.avsc merged/commons/passive/phone/phone_sms_unread.avsc merged/commons/passive/phone/phone_contact_list.avsc merged/commons/passive/empatica/empatica_e4_battery_level.avsc merged/commons/passive/empatica/empatica_e4_electrodermal_activity.avsc merged/commons/passive/empatica/empatica_e4_tag.avsc merged/commons/passive/empatica/empatica_e4_acceleration.avsc merged/commons/passive/empatica/empatica_e4_blood_volume_pulse.avsc merged/commons/passive/empatica/empatica_e4_inter_beat_interval.avsc merged/commons/passive/empatica/empatica_e4_sensor_status.avsc merged/commons/passive/empatica/empatica_e4_temperature.avsc merged/commons/passive/biovotion/biovotion_vsm1_ppg_raw.avsc merged/commons/passive/biovotion/biovotion_vsm1_led_current.avsc merged/commons/passive/biovotion/biovotion_vsm1_galvanic_skin_response.avsc merged/commons/passive/biovotion/biovotion_vsm1_temperature.avsc merged/commons/passive/biovotion/biovotion_vsm1_blood_pulse_wave.avsc merged/commons/passive/biovotion/biovotion_vsm1_battery_level.avsc merged/commons/passive/biovotion/biovotion_vsm1_heart_rate_variability.avsc merged/commons/passive/biovotion/biovotion_vsm1_energy.avsc merged/commons/passive/biovotion/biovotion_vsm1_oxygen_saturation.avsc merged/commons/passive/biovotion/biovotion_vsm1_acceleration.avsc merged/commons/passive/biovotion/biovotion_vsm1_respiration_rate.avsc merged/commons/passive/biovotion/biovotion_vsm1_heart_rate.avsc
Dependent: merged/commons/stream/aggregator/aggregate_list.avsc merged/commons/active/thincit/thinc_it_pdq.avsc merged/commons/active/thincit/thinc_it_symbol_check.avsc merged/commons/active/thincit/thinc_it_spotter.avsc merged/commons/active/thincit/thinc_it_code_breaker.avsc
```

**Comment by yatharthranjan** [comment]
As a temporary fix I am just removing all the dependent files from the notdependent as follows- 
```sh
for i in "${dependent[@]}"; do
         notdependent=(${notdependent[@]//*$i*})
done
```

**Comment by yatharthranjan** [comment]
Fixed by #116 and #117
